### PR TITLE
Refactor add goal flow

### LINKED ISSUE #72: Polish Add new Goal View

### PR DESCRIPTION
**Pull request by josemontesdeoca**
Refactor Add a New Goal flow based on the design.

This resolves #72 
@fcduarte @amitbharadwaj please review.

### CODE REVIEW

**Comment by fcduarte** [comment]
This branch is two commits behind. Please update it.

File: src/org/missionassetfund/apps/android/fragments/NameInputFragment.java [hunk 1]
```diff
@@ -32,6 +33,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         btnBack = (ImageButton) view.findViewById(R.id.btnBack);
         btnNext = (ImageButton) view.findViewById(R.id.btnNext);
 
+        // Ugly hack to try reuse this view in both goal / transaction
```
Comment (fcduarte) [inline src/org/missionassetfund/apps/android/fragments/NameInputFragment.java]: When you update this branch I'll be able to access this field from the listener.. Then you could change this code at the activity.

File: src/org/missionassetfund/apps/android/fragments/AmountInputFragment.java [hunk 2]
```diff
@@ -33,6 +36,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         etAmount = (EditText) view.findViewById(R.id.etAmount);
         rbExpense = (RadioButton) view.findViewById(R.id.rbExpense);
         btnNext = (ImageButton) view.findViewById(R.id.btnNext);
+        rgType = (RadioGroup) view.findViewById(R.id.rgType);
+
+        // Ugly hack to try reuse this view in both goal / transaction
```
Comment (fcduarte) [inline src/org/missionassetfund/apps/android/fragments/AmountInputFragment.java]: Create a listener for inside `onCreateView` then on the activity that is implementing it you could make this change there.

**Comment by josemontesdeoca** [comment]
@fcduarte I believe I understood your feedback and removed the UGLY hack in favor of the standar listener pattern. Thanks for the guidance here :+1: . Please review.

**Comment by fcduarte** [comment]
@josemontesdeoca cool, looks much more cleaner! :shipit:

**Comment by josemontesdeoca** [comment]
Indeed. Thanks for the help. ;)

**Pull request merged**